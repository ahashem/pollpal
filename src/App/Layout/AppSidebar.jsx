import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Layout, Menu } from 'antd';

const { Sider } = Layout;

/**
 * Application Sidebar wrapper
 * @param {Array} children
 * @param {boolean} collapsed
 * @param {function} onCollapse
 */
const AppSidebar = ({ children, collapsed, onCollapse }) => (
  <Sider
    collapsible
    collapsed={collapsed}
    onCollapse={onCollapse}
  >
    {/*Dummy Menu - Actually no use of it but a simple style!*/}
    <Menu theme="dark" mode="inline" defaultSelectedKeys={['1']}>
      <Menu.Item key="1">
        <Icon type="dashboard"/>
        <span>Dashboard</span>
      </Menu.Item>
    </Menu>
    {children}
  </Sider>
);

AppSidebar.propTypes = {
  children: PropTypes.node,
  collapsed: PropTypes.bool,
  onCollapse: PropTypes.func
};

AppSidebar.defaultProps = {
  children: [],
  collapsed: false,
  onCollapse: () => {}
};

export default AppSidebar;
