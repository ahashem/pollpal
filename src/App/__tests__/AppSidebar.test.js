import React from 'react';
import { shallow } from 'enzyme/build';
import AppSidebar from '../Layout/AppSidebar';

describe('<AppSidebar /> component', () => {
  it('should render without children', () => {
    const rendered = shallow(
      <AppSidebar />
    );

    expect(rendered).toMatchSnapshot();
  });

  it('should render with children', () => {
    const mockChildren = [];
    const onCollapse = jest.fn();

    const rendered = shallow(
      <AppSidebar collapsed={false} onCollapse={onCollapse}>
        {mockChildren}
      </AppSidebar>
    );

    expect(rendered).toMatchSnapshot();
  });
});
