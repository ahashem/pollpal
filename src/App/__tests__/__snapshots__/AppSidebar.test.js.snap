// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppSidebar /> component should render with children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppSidebar
    collapsed={false}
    onCollapse={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Menu
          className=""
          defaultSelectedKeys={
            Array [
              "1",
            ]
          }
          focusable={false}
          mode="inline"
          prefixCls="ant-menu"
          theme="dark"
        >
          <MenuItem>
            <Icon
              type="dashboard"
            />
            <span>
              Dashboard
            </span>
          </MenuItem>
        </Menu>,
        Array [],
      ],
      "collapsed": false,
      "collapsedWidth": 80,
      "collapsible": true,
      "defaultCollapsed": false,
      "onCollapse": [MockFunction],
      "prefixCls": "ant-layout-sider",
      "reverseArrow": false,
      "style": Object {},
      "theme": "dark",
      "width": 200,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <MenuItem>
            <Icon
              type="dashboard"
            />
            <span>
              Dashboard
            </span>
          </MenuItem>,
          "className": "",
          "defaultSelectedKeys": Array [
            "1",
          ],
          "focusable": false,
          "mode": "inline",
          "prefixCls": "ant-menu",
          "theme": "dark",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Icon
                type="dashboard"
              />,
              <span>
                Dashboard
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "type": "dashboard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dashboard",
              },
              "ref": null,
              "rendered": "Dashboard",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Menu
            className=""
            defaultSelectedKeys={
              Array [
                "1",
              ]
            }
            focusable={false}
            mode="inline"
            prefixCls="ant-menu"
            theme="dark"
          >
            <MenuItem>
              <Icon
                type="dashboard"
              />
              <span>
                Dashboard
              </span>
            </MenuItem>
          </Menu>,
          Array [],
        ],
        "collapsed": false,
        "collapsedWidth": 80,
        "collapsible": true,
        "defaultCollapsed": false,
        "onCollapse": [MockFunction],
        "prefixCls": "ant-layout-sider",
        "reverseArrow": false,
        "style": Object {},
        "theme": "dark",
        "width": 200,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MenuItem>
              <Icon
                type="dashboard"
              />
              <span>
                Dashboard
              </span>
            </MenuItem>,
            "className": "",
            "defaultSelectedKeys": Array [
              "1",
            ],
            "focusable": false,
            "mode": "inline",
            "prefixCls": "ant-menu",
            "theme": "dark",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Icon
                  type="dashboard"
                />,
                <span>
                  Dashboard
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "type": "dashboard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Dashboard",
                },
                "ref": null,
                "rendered": "Dashboard",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<AppSidebar /> component should render without children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppSidebar
    collapsed={false}
    onCollapse={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Menu
          className=""
          defaultSelectedKeys={
            Array [
              "1",
            ]
          }
          focusable={false}
          mode="inline"
          prefixCls="ant-menu"
          theme="dark"
        >
          <MenuItem>
            <Icon
              type="dashboard"
            />
            <span>
              Dashboard
            </span>
          </MenuItem>
        </Menu>,
        Array [],
      ],
      "collapsed": false,
      "collapsedWidth": 80,
      "collapsible": true,
      "defaultCollapsed": false,
      "onCollapse": [Function],
      "prefixCls": "ant-layout-sider",
      "reverseArrow": false,
      "style": Object {},
      "theme": "dark",
      "width": 200,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <MenuItem>
            <Icon
              type="dashboard"
            />
            <span>
              Dashboard
            </span>
          </MenuItem>,
          "className": "",
          "defaultSelectedKeys": Array [
            "1",
          ],
          "focusable": false,
          "mode": "inline",
          "prefixCls": "ant-menu",
          "theme": "dark",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Icon
                type="dashboard"
              />,
              <span>
                Dashboard
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "type": "dashboard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dashboard",
              },
              "ref": null,
              "rendered": "Dashboard",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Menu
            className=""
            defaultSelectedKeys={
              Array [
                "1",
              ]
            }
            focusable={false}
            mode="inline"
            prefixCls="ant-menu"
            theme="dark"
          >
            <MenuItem>
              <Icon
                type="dashboard"
              />
              <span>
                Dashboard
              </span>
            </MenuItem>
          </Menu>,
          Array [],
        ],
        "collapsed": false,
        "collapsedWidth": 80,
        "collapsible": true,
        "defaultCollapsed": false,
        "onCollapse": [Function],
        "prefixCls": "ant-layout-sider",
        "reverseArrow": false,
        "style": Object {},
        "theme": "dark",
        "width": 200,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MenuItem>
              <Icon
                type="dashboard"
              />
              <span>
                Dashboard
              </span>
            </MenuItem>,
            "className": "",
            "defaultSelectedKeys": Array [
              "1",
            ],
            "focusable": false,
            "mode": "inline",
            "prefixCls": "ant-menu",
            "theme": "dark",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Icon
                  type="dashboard"
                />,
                <span>
                  Dashboard
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "type": "dashboard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Dashboard",
                },
                "ref": null,
                "rendered": "Dashboard",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
